src/decorate.test.ts

```diff
@@ -157,12 +157,12 @@ test('decorate correct target for function', () => {
 
 // DecoratorCorrectNameInPipelineForPropertyOverload
 test('decorate correct target for function', () => {
-  const sent: string[] = [];
+  const sent: PropertyKey[] = [];
   const decorators = [
-    (target: object, name: string) => { sent.push(name); },
-    (target: object, name: string) => { sent.push(name); },
-    (target: object, name: string) => { sent.push(name); },
-    (target: object, name: string) => { sent.push(name); },
+    (_target: object, name: PropertyKey) => { sent.push(name); },
+    (_target: object, name: PropertyKey) => { sent.push(name); },
+    (_target: object, name: PropertyKey) => { sent.push(name); },
+    (_target: object, name: PropertyKey) => { sent.push(name); },
```

Comment:
`_target` is an opt-out from noUnusedParameters check: https://github.com/Microsoft/TypeScript/issues/9458#issuecomment-229840297

See also tslint rules adjustment bellow